### PR TITLE
Evaluating the current size for protected cache

### PR DESCRIPTION
**Pull request by andrey-kashcheev**
Added ability to get current size of the DiskCache. For protected cache
this value is pretty precise, for mutable cache it will depend on the
leveldb internals (not recommended to use for mutable cache)

Relates-To: OLPEDGE-2393
Signed-off-by: Andrey Kashcheev <ext-andrey.kashcheev@here.com>

### CODE REVIEW

File: external/boost/CMakeLists.txt.boost.in [hunk 1]
```diff
@@ -30,6 +30,7 @@ ExternalProject_Add(boost-download
                         libs/container_hash
                         libs/core
                         libs/detail
+                        libs/filesystem
```
Comment (andescu) [inline external/boost/CMakeLists.txt.boost.in:33]: This is not good, that we take in extra boost filesystem library just for tests. Even now we have a lot of dependencies and checking out boost repo takes ages. Maybe we can add it only when tests are enabled. Is this possible?

File: external/boost/CMakeLists.txt.boost.in [hunk 2]
```diff
@@ -58,7 +59,7 @@ ExternalProject_Add(boost-download
     GIT_SHALLOW         1
     SOURCE_DIR          "@CMAKE_CURRENT_BINARY_DIR@/external_boost"
     UPDATE_COMMAND      ""
-    CONFIGURE_COMMAND   cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @BOOTSTRAP_CMD@ && cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @B2_CMD@ headers
+    CONFIGURE_COMMAND   cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @BOOTSTRAP_CMD@ && cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @B2_CMD@
```
Comment (andescu) [inline external/boost/CMakeLists.txt.boost.in:62]: Same here. If we do it like this then we will need to compile boost during cmake init. Please do it only if tests are enabled.

File: olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp [hunk 3]
```diff
@@ -828,6 +828,10 @@ DefaultCacheImpl::GetFromDiscCache(const std::string& key) {
   return boost::none;
 }
 
+uint64_t DefaultCacheImpl::GetProtectedCacheSize() const {
+  return protected_cache_->Size();
```
Comment (andescu) [inline olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp:832]: Please check that `protected_cache_` is not `nullptr` before. There is not always a protected cache set.

File: olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp [hunk 4]
```diff
@@ -828,6 +828,10 @@ DefaultCacheImpl::GetFromDiscCache(const std::string& key) {
   return boost::none;
 }
 
+uint64_t DefaultCacheImpl::GetProtectedCacheSize() const {
+  return protected_cache_ ? protected_cache_->Size() : 0;
```
Comment (andescu) [inline olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp:832]:
```suggestion
  return protected_cache_ ? protected_cache_->Size() : 0u;
```

File: olp-cpp-sdk-core/include/olp/core/utils/Dir.h [hunk 5]
```diff
@@ -105,6 +106,19 @@ class CORE_API Dir {
...
+   * unnecessary files or directories from calculation.
+   *
+   * @param path Path of the directory.
+   * @param filter_fn Filter function.
+   *
+   * @return Calculated size.
+   */
+  static uint64_t Size(
+      const std::string& path,
+      std::function<bool(const std::string& /*path*/)> filter_fn = nullptr);
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/utils/Dir.h:121]: Maybe add an alias for the lambda.

**Review by mykhailo-kuchma — DISMISSED** [review]

**Review by meravingen123 — APPROVED** [review]

**Review by ystefinko — APPROVED** [review]

**Review by kzvieriev — APPROVED** [review]

**Pull request merged**